asserts/system_user.go

```diff
@@ -216,6 +246,30 @@ func checkHashedPassword(headers map[string]interface{}, name string) (string, e
 	return pw, nil
 }
 
+func parseSystemUserExpiration(headers map[string]interface{}, until time.Time) (string, error) {
+	value, ok := headers["user-valid-for"]
+	if !ok {
+		return "", nil
+	}
+
+	str, ok := value.(string)
+	if !ok {
+		return "", fmt.Errorf("cannot parse 'user-valid-for': not a string")
+	}
+
+	if strings.ToLower(str) == "until-expiration" {
+		return str, nil
+	}
+
+	// If the value wasn't 'until-expiration' then the value must
+	// be a duration.
+	_, err := checkDuration(headers, "user-valid-for")
+	if err != nil {
+		return "", fmt.Errorf("cannot parse 'user-valid-for': %q is invalid", str)
```

Comment:
this wrapping is unsual as as the check function should produce complete errors itself

Reply:
This is now removed